src/gc/collection.rs

```diff
@@ -137,16 +133,29 @@ async unsafe fn process_mutation_buffer(
     mutation_buffer_rx: &mut UnboundedReceiver<Mutation>,
     mutation_buffer: &mut Vec<Mutation>,
 ) {
+    // [std::cmp::clamp] has an assertion where min must be larger than max, so this cannot be done
+    // using [std::cmp::clamp]
+    #[expect(clippy::manual_clamp)]
     // It is very important that we do not delay any mutations that
     // have occurred at this point by an extra epoch.
-    let to_recv = mutation_buffer_rx.len();
-    mutation_buffer_rx.recv_many(mutation_buffer, to_recv).await;
+    let mut to_recv = mutation_buffer_rx
+        .len()
+        .min(MAX_MUTATIONS_PER_EPOCH)
+        .max(MIN_MUTATIONS_PER_EPOCH);
+
+    loop {
+        mutation_buffer_rx.recv_many(mutation_buffer, to_recv).await;
+        for mutation in mutation_buffer.drain(..) {
+            match mutation.kind {
+                MutationKind::Inc => increment(mutation.gc),
+                MutationKind::Dec => decrement(mutation.gc),
+            }
+        }
 
-    for mutation in mutation_buffer {
-        match mutation.kind {
-            MutationKind::Inc => increment(mutation.gc),
-            MutationKind::Dec => decrement(mutation.gc),
+        if mutation_buffer_rx.is_empty() {
+            break;
         }
+        to_recv = mutation_buffer_rx.len().min(MAX_MUTATIONS_PER_EPOCH);
```

Comment:
I think the issue with this method, i.e. capping the maximum number of mutations per loop, is that in the case there are a LOT of mutations constantly, is that this loop will never exit and we'll never free any data.